File: C++/12-数值的整数次方.cpp
class Solution {
public:
    double Power(double base, int exponent) {
        if(base == 0)
            return 0;
        if(exponent == 0)
            return 1;
        if(exponent == 1)
            return base;
        int e = abs(exponent);
        double res = 1.0;
        while(e){
            if(e & 1)
                res *= base;
            base *= base;
            e = e >> 1;
        }
        return exponent < 0 ? 1 / res : res;
    }
};

File: Python/12-数值的整数次方.py
class Solution:
    def Power(self, base, exponent):
        # write code here
        if not base:
            return 0
        if not exponent:
            return 1
        if exponent == 1:
            return base
        e = abs(exponent)
        res = 1
        while e:
            if e & 1:
                res *= base
            base *= base
            e = e >> 1
        return res if exponent > 0 else 1 / res
